internal/controllers/cluster/cluster_controller_status_test.go

```diff
@@ -765,29 +765,48 @@ func TestSetWorkersAvailableCondition(t *testing.T) {
 	}
 }
 
-func TestSetMachinesReadyCondition(t *testing.T) {
+func TestSetControlPlaneMachinesReadyCondition(t *testing.T) {
```

Comment:
I think we should have separated tests for all the 4 conditions to validate all the combinations (with the proposed approach we are going to test only partially the control plane part)